coverage/PENDING_REVIEW/intel/0424-COVERAGE-nir-builder-h-724/cov-vector-illegal-index-never-executed.amber

```diff
@@ -0,0 +1,237 @@
+#!amber
+
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+# A test for a coverage-gap found by the GraphicsFuzz project.
+
+# Short description: A fragment shader that covers specific NIR code paths
+
+# The test passes because the shader always writes red.
+
+# Optimized using spirv-opt with the following arguments:
+# '--simplify-instructions'
+# '--reduce-load-size'
+# '--inline-entry-points-exhaustive'
+# '--eliminate-local-single-block'
+# '--redundancy-elimination'
+# '--eliminate-dead-branches'
+# '--eliminate-dead-inserts'
+# '--redundancy-elimination'
+# '--eliminate-local-single-block'
+# '--eliminate-local-multi-store'
+# '--simplify-instructions'
+# '--eliminate-local-multi-store'
+# '--eliminate-dead-branches'
+# '--eliminate-dead-branches'
+# '--merge-return'
+# '--combine-access-chains'
+# '--eliminate-local-multi-store'
+# '--eliminate-local-single-block'
+# '--eliminate-local-single-store'
+# '--eliminate-dead-branches'
+# '--merge-blocks'
+# '--eliminate-dead-branches'
+# '--merge-blocks'
+# '--simplify-instructions'
+# '--eliminate-dead-branches'
+# '--merge-blocks'
+# '--eliminate-local-multi-store'
+# '--eliminate-dead-branches'
+# '--merge-return'
+# '--ccp'
+# '--eliminate-local-single-store'
+# '--redundancy-elimination'
+# '--eliminate-local-multi-store'
+# '--eliminate-dead-code-aggressive'
+# spirv-opt commit hash: a0370efd589be33d5d9a85cfde2f85841b3755af
+
+
+
+SHADER vertex variant_vertex_shader PASSTHROUGH
+
+# variant_fragment_shader is derived from the following GLSL:
+# #version 320 es
+# #define _int_0 _GLF_uniform_int_values[0]
+# #define _int_1 _GLF_uniform_int_values[1]
+# #define _float_1_0 _GLF_uniform_float_values[0]
+#
+# precision highp int;
+# precision highp float;
+#
+# // Contents of _GLF_uniform_float_values: 1.0
+# layout(set = 0, binding = 0) uniform buf0
+# {
+#     float _GLF_uniform_float_values[1];
+# };
+#
+# // Contents of _GLF_uniform_int_values: [0, 1]
+# layout(set = 0, binding = 1) uniform buf1
+# {
+#     int _GLF_uniform_int_values[2];
+# };
+#
+# // Contents of zero: 0
+# layout(set = 0, binding = 2) uniform buf2
+# {
+#     int zero;
+# };
+#
+# layout(location = 0) out vec4 _GLF_color;
+#
+# void main()
+# {
+#     vec3 v = vec3(_int_1);
+#     int a = 3;
+#
+#     if(zero > _int_0)
+#     {
+#         v[a] = _float_1_0;
+#     }
+#
+#     _GLF_color = vec4(v[_int_1], _int_0, _int_0, _int_1);
```

Comment:
Could `a` be set to 1? And could `_float_1_0` be 0? I.e. can we make it so, if the if WAS executed then the test would fail? If this doesn't, go ahead and add a comment so that, if we review again, we know there is nothing more we can do with this test without missing the covpoint.

Also: can we add an "always false" comment above the if?

Reply:
a needs to be 3 to get an out of bounds access. I've added a comment above line 101 to make it clear.

Reply:
@paulthomson Are you happy to re-review?

Reply:
Makes sense. Can we add a comment to make it clear that this is required? Something like: statically out of bound access is required to cover this cov point.